client/components/data/query-jetpack-sale-coupon/index.ts

```diff
@@ -1,17 +1,21 @@
 import { useEffect } from 'react';
 import { useDispatch, useSelector } from 'react-redux';
 import { fetchJetpackSaleCoupon } from 'calypso/state/marketing/actions';
-import { getJetpackSaleCoupon } from 'calypso/state/marketing/selectors';
+import {
+	getHasRequestedJetpackSaleCoupon,
+	getIsRequestingJetpackSaleCoupon,
+} from 'calypso/state/marketing/selectors';
 
 export default function QueryJetpackSaleCoupon(): null {
 	const dispatch = useDispatch();
-	const saleCoupon = useSelector( getJetpackSaleCoupon );
+	const hasRequestedCoupon = useSelector( getHasRequestedJetpackSaleCoupon );
+	const isRequestingCoupon = useSelector( getIsRequestingJetpackSaleCoupon );
 
 	useEffect( () => {
-		if ( ! saleCoupon ) {
+		if ( ! isRequestingCoupon && ! hasRequestedCoupon ) {
```

Comment:
`isRequesting` and `hasRequested` are distinct to better inform `<IntroPricingBanner />` loading state. If we were just to use `isRequesting` there is the chance of a "flash" of the text before the request starts. With `hasRequested` it will not render until the request has finished.